service_packs/kubernetes/general/general.go

```
@@ -1,352 +1,316 @@
+	// Create a curl command to access the supplied url and only show http response in stdout.
+	cmd := "curl -s -o /dev/null -I -L -w %{http_code} " + urlAddress
 
-	if err != nil {
-		err = utils.ReformatError("[ERROR] Error raised when attempting to access URL: %v", err)
-		log.Print(err)
-	}
+	stepTrace.WriteString(fmt.Sprintf("Attempt to run command in the pod: '%s'; ", cmd))
+	exitCode, stdOut, cmdErr := conn.ExecCommand(cmd, scenario.namespace, scenario.pods[0])
 
-	//hold on to the code
-	s.httpStatusCode = code
+	// Validate that no internal error occurred during execution of curl command
+	if cmdErr != nil && exitCode == -1 {
+		err = utils.ReformatError("Error raised when attempting to execute curl command inside container: %v", cmdErr)
+		return err
+	}
 
-	stepTrace.WriteString(fmt.Sprintf(
-		"Proces inside the pod established http connection with url '%s',", url))
-	payload = struct {
-		PodState kubernetes.PodState
-	}{s.podState}
+	// TODO: Confirm this implementation:
+	// Expected Exit Code from Curl command is:	6
+	// Expected HTTP Response in stdout is:		000
 
-	return err
-}
+	stepTrace.WriteString("Check expected exit code was raised from curl command; ")
+	if exitCode != expectedCurlExitCode {
+		err = utils.ReformatError("Unexpected exit code: %d Error: %v", exitCode, cmdErr)
+		return err
+	}
 
-func (s *scenarioState) accessIs(accessResult string) error {
-	// Standard auditing logic to ensures panics are also audited
-	stepTrace, payload, err := utils.AuditPlaceholders()
-	defer func() {
-		s.audit.AuditScenarioStep(s.currentStep, stepTrace.String(), payload, err)
-	}()
+	// Validate that stdout contains valid http response code
+	httpResponse, parseErr := strconv.Atoi(stdOut)
+	if parseErr != nil {
+		err = utils.ReformatError("Unexpected result in stdout. Please ensure curl command is in the following format so that only http response code is added to standard output: '%s'", "curl -s -o /dev/null -I -L -w %%{http_code} _urlAddress_")
```

Perhaps this can be left as a comment warning? Then we could just return the strconv error if present